shell.client/src/main/java/org/metaborg/spoofax/shell/client/Repl.java

```diff
@@ -0,0 +1,96 @@
+package org.metaborg.spoofax.shell.client;
+
+import org.fusesource.jansi.Ansi;
+import org.fusesource.jansi.Ansi.Color;
+import org.metaborg.spoofax.shell.commands.ICommandInvoker;
+import org.metaborg.spoofax.shell.commands.SpoofaxCommandFactory;
+import org.metaborg.spoofax.shell.commands.SpoofaxCommandInvoker;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+
+/**
+ * Interactive REPL (Read-Eval-Print Loop) which reads an expression or command typed in by the user
+ * via an {@link IEditor}, executes them with a {@link ICommandInvoker} and prints it to an
+ * {@link IDisplay}.
+ */
+public final class Repl {
+    private ICommandInvoker invoker;
+    private IEditor editor;
+    private IDisplay display;
+
+    /**
+     * @param in
+     *            The {@link InputStream} from which to read user input
+     * @param out
+     *            The {@link PrintStream} to write results to.
+     * @param err
+     *            The {@link PrintStream} to write errors to.
+     * @throws IOException
+     *             when an IO error occurs.
+     */
+    public Repl(InputStream in, PrintStream out, PrintStream err) throws IOException {
+        this(new SpoofaxCommandInvoker(), in, out, err);
+    }
+
+    /**
+     * @param invoker
+     *            The {@link ICommandInvoker} for executing user input.
+     * @param in
+     *            The {@link InputStream} from which to read user input
+     * @param out
+     *            The {@link PrintStream} to write results to.
+     * @param err
+     *            The {@link PrintStream} to write errors to.
+     * @throws IOException
+     *             when an IO error occurs.
+     */
+    public Repl(ICommandInvoker invoker, InputStream in, OutputStream out, OutputStream err)
+        throws IOException {
+        this.invoker = invoker;
+        TerminalUserInterface ui = new TerminalUserInterface(in, out, err);
+        ui.setPrompt(coloredFg(Color.RED, "[In ]: "));
```

Comment:
Small nitpick, but I (and I think others too) associate red with error, so perhaps the default color should be changed to something without an associated meaning. Also, I think we should stick with symbols only, unless we start to get more than just singleline/multiline prompts (my suggestion is to follow Python and use `>>>` for singleline and `...` for multiline)